sys/include/net/gnrc/lwmac/lwmac_init.h

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2015 Daniel Krebs
+ *               2016 INRIA
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_lwmac
+ * @{
+ *
+ * @file
+ * @brief       Initialization of LWMAC
+ * @internal
+ * @author      Daniel Krebs <github@daniel-krebs.net>
+ * @author      Shuguo Zhuo  <shuguo.zhuo@inria.fr>
+ */
+
+#ifndef GNRC_LWMAC_INIT_H
+#define GNRC_LWMAC_INIT_H
+
+#include "net/gnrc/netdev.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Initialize an instance of the LWMAC layer
+ *
+ * The initialization starts a new thread that connects to the given netdev
+ * device and starts a link layer event loop.
+ *
+ * @param[in] stack         stack for the control thread
+ * @param[in] stacksize     size of *stack*
+ * @param[in] priority      priority for the thread housing the LWMAC instance
+ * @param[in] name          name of the thread housing the LWMAC instance
+ * @param[in] dev           netdev device, needs to be already initialized
+ *
+ * @return                  PID of LWMAC thread on success
+ * @return                  -EINVAL if creation of thread fails
+ * @return                  -ENODEV if *dev* is invalid
+ */
+kernel_pid_t gnrc_lwmac_init(char *stack, int stacksize, char priority,
```

Comment:
@miri64 I found it is hard to decompose:  the entity of `lwmac` (`lwmac_t lwmac;`) is put in the structure of gnrc_netdev in `net/gnrc/netdev.h` which means `net/gnrc/netdev.h` must includes the header files defines `lwmac_t` (`lwmac/types.h` or `lwmac/lwmac.h`). No matter where I put the definition of `lwmac_t`, in `lwmac/types.h` or `lwmac/lwmac.h`, that all leads to cyclic dependency. For instance, if `lwmac_t` is defined in `lwmac/types.h`, `lwmac/types.h` has to call `lwmac/lwmac.h` that has all Lw-MAC macros to define the number of timeouts defined in `lwmac_t` (the old problem). If `lwmac_t` is defined in `lwmac/lwmac.h`, then, `net/gnrc/netdev.h` must call `lwmac/lwmac.h`, sure cyclic inclusion. 

The solution maybe to remove `lwmac_t lwmac` from the structure of `gnrc_netdev_t` and place it in `lwmac.c`, like `static lwmac_t lwmac;` to make a entity in in `lwmac.c`. But that will lead to too many interfaces changes (may affect near all functions of Lw-MAC). I don't quite like to do such a huge change... :-/